### PR TITLE
expose haproxy prometheus

### PR DESCRIPTION
**Pull request by pdgetrf**
### Changes

Patch Haproxy to expose Prometheus metrics

### Validation
- The patching part was tested on a local box
- Verified 100-node cluster started up and both apiserver and haproxy metrics showed up in prometheus

### CODE REVIEW

File: cluster/gce/util.sh [hunk 1]
```diff
@@ -2852,11 +2852,26 @@ function setup-proxy() {
...
+  patch-haproxy-prometheus
+
   ssh-to-node ${PROXY_NAME} "sudo sed -i '/^ExecStart=.*/a ExecStartPost=/bin/bash -c \"sleep 20 && for npid in \$(pidof ${KUBERNETES_SCALEOUT_PROXY_APP}); do sudo prlimit --pid \$npid --nofile=500000:500000 ; done\"' /lib/systemd/system/${KUBERNETES_SCALEOUT_PROXY_APP}.service"
   ssh-to-node ${PROXY_NAME} "sudo systemctl daemon-reload"
   ssh-to-node ${PROXY_NAME} "sudo systemctl restart ${KUBERNETES_SCALEOUT_PROXY_APP}"
 }
 
+function patch-haproxy-prometheus {
+  # based on https://www.haproxy.com/blog/haproxy-exposes-a-prometheus-metrics-endpoint
+  ssh-to-node ${PROXY_NAME} "echo 'Patching Haproxy to expose prometheus...'"
```
Comment (chenqianfzh) [inline cluster/gce/util.sh:2864]: Minor: Run "echo 'Patching Haproxy to expose prometheus...'" directly on the test machine instead of the proxy machine?

**Review by chenqianfzh — APPROVED** [review]

File: cluster/gce/util.sh (hunk 1 above)
Comment (pdgetrf) [reply cluster/gce/util.sh]: fixed

**Pull request merged**